readthedocs/projects/tasks.py

```diff
@@ -203,8 +203,11 @@ def validate_duplicate_reserved_versions(self, data):
 @app.task(max_retries=5, default_retry_delay=7 * 60)
 def sync_repository_task(version_pk):
     """Celery task to trigger VCS version sync."""
-    step = SyncRepositoryTaskStep()
-    return step.run(version_pk)
+    try:
+        step = SyncRepositoryTaskStep()
+        return step.run(version_pk)
+    finally:
+        clean_build_task(version_pk)
```

Comment:
With this implementation this `clean_build_task` is not a task anymore.

I think we could,

1. remove the `_task` from the name
1. remove the decorator around the function definition
1. move it to another file where other utils for the tasks live

Reply:
I wanted to keep it as a task just in case we want to execute that from the command line, but we can do that later if needed.